File: backend/src/main/java/com/messmate/backend/dto/response/MealSelectionDashboardResponse.java
package com.messmate.backend.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MealSelectionDashboardResponse {
    private Integer advanceBookingDays;
    private String lunchVotingDeadline;
    private String dinnerVotingDeadline;
    private String currentServerTime;

    // Month Summary
    private int currentMonthTotalMeals;
    private int currentMonthLunchCount;
    private int currentMonthDinnerCount;

    // History & Future Data
    private List<MealStatusResponse> recentHistory; // Last 5 days
    private List<MealStatusResponse> futureSelections; // Today to Today + limit
}

File: android/app/src/main/java/com/messmate/android/data/meal/MealSelectionDashboardResponse.kt
package com.messmate.android.data.meal

data class MealSelectionDashboardResponse(
    val advanceBookingDays: Int,
    val lunchVotingDeadline: String,
    val dinnerVotingDeadline: String,
    val currentServerTime: String,
    val currentMonthTotalMeals: Int,
    val currentMonthLunchCount: Int,
    val currentMonthDinnerCount: Int,
    val recentHistory: List<MealStatusResponse>,
    val futureSelections: List<MealStatusResponse>
)
